src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -1,116 +1,528 @@
+import { BasicButton } from 'components/Form';
+import { PhotoCamera } from '@material-ui/icons';
+import { Autocomplete } from '@material-ui/lab';
+import { uploadImageToImgur } from 'old/lib/utilities/Imgur/uploadImageToImgur';
+import { getStringFromFile } from 'old/lib/utilities/Imgur/getStringFromFile';
+import {
+  getCitiesByRegionId,
+  getRegionByCityId,
+} from 'old/lib/utilities/API/api';
 import { useStyles } from './styles/PersonalInfo.styles';
-import { IExpert } from '../../../old/lib/types';
-import { getCurrentUser } from '../../../old/lib/utilities/API/api';
-import { ERROR_404 } from '../../../old/lib/constants/routes';
+import { ICity, IRegion } from '../../../old/lib/types';
+import { ErrorField } from './ErrorField';
+import { regex } from './constants/regex';
+import { fields } from './constants/fields';
+import { INewAuthorValues } from './types';
 
 export const PersonalInfo: React.FC = () => {
   const classes = useStyles();
-  const { expertId } = useParams<{ expertId: string }>();
-  const [loadedExpert, setLoadedExpert] = useState<IExpert>();
-  const history = useHistory();
-
-  const fetchExpert = useCallback(async () => {
-    try {
-      const expertResponse = await getCurrentUser();
-      setLoadedExpert(expertResponse.data);
-    } catch (error) {
-      history.push(ERROR_404);
+  const regions = useSelector(
+    (state: RootStateType) => state.properties.regions,
+  );
+  const cities = useSelector((state: RootStateType) => state.properties.cities);
+  const [selectRegions, setSelectRegions] = useState<IRegion[] | null>(null);
+  const [regionForValue, setRegionForValue] = useState<IRegion | null>(null);
+  const [selectCities, setSelectCities] = useState<ICity[] | null>();
+  const [isCitySelect, setIsCitySelect] = useState<boolean>(false);
+  const [newAuthorValues, setNewAuthorValues] = useState<INewAuthorValues>({
+    avatar: '',
+    firstName: '',
+    lastName: '',
+    regionId: 0,
+    cityId: 0,
+    bio: '',
+    email: '',
+    /* eslint-disable-next-line */
+    socialNetwork: new Array(5).fill(null),
+  });
+  const [errorFields, setErrorFields] = useState({
+    avatar: fields.avatarRequired,
+    lastName: fields.emptyField,
+    firstName: fields.emptyField,
+    regionId: fields.emptyField,
+    cityId: fields.emptyField,
+    email: fields.emptyField,
+    work: fields.emptyField,
+    bio: fields.emptyField,
+    facebook: '',
+    instagram: '',
+    youtube: '',
+    twitter: '',
+    linkedin: '',
+    socialNetwoks: fields.socialNetworksEmpty,
+  });
+  const [visitFields, setVisitFields] = useState({
+    lastName: false,
+    firstName: false,
+    region: false,
+    city: false,
+    email: false,
+    work: false,
+    bio: false,
+    facebook: false,
+    instagram: false,
+    youtube: false,
+    twitter: false,
+    linkedin: false,
+  });
+  const [toggleButton, setToggleButton] = useState(true);
+
+  const setRegionByCityId = async (id: number) => {
```

Comment:
may it make sense to move region picker with all this fetching thing into separate component?

Reply:
cool!